### PR TITLE
altered the ellipsis dropdown to be more flexible, added ellipsis dro…

### PR DESCRIPTION
**Pull request by HankC138**
…pdown to assignment in mystaffplan

added ellipsis menu on individual assignment in mystaffplan. If the delete assignment functionality is approved I will wire up the delete option as well. For now it is a simple modal to alter the dates, hours, and status of an existing assignment 

also reformatted the ellipsisDropdown menu to be more reusable

### CODE REVIEW

File: app/components/ellipsisDropdownMenu.tsx [hunk 1]
```diff
@@ -1,21 +1,25 @@
-import { Fragment } from "react";
-import { Menu, Transition } from "@headlessui/react";
+import { Fragment, ReactNode } from "react";
+import { Menu,MenuItem, MenuButton, MenuItems, Transition } from "@headlessui/react";
 import React from "react";
-import Link from "next/link";
 
 interface DropdownMenuProps {
-    options: { label: string, href: string, show?: boolean }[];
+    options: {  component:()=>ReactNode,show?: boolean }[];
```
Comment (zhukovdigital) [inline app/components/ellipsisDropdownMenu.tsx:6]: Just change the props here to apply a component without func. This comment related to my comments below.

Comment (HankC138) [reply app/components/ellipsisDropdownMenu.tsx:6]: fixed

File: app/components/ellipsisDropdownMenu.tsx [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
+    location: 'navbar' | 'userAssignment';
 }
 
-const EllipsisDropdownMenu: React.FC<DropdownMenuProps> = ({ options }) => {
+const EllipsisDropdownMenu: React.FC<DropdownMenuProps> = ({ options, location }) => {
+    const ellipsisENUM = {
```
[the remark this answers is not in the document]
Comment (HankC138) [reply app/components/ellipsisDropdownMenu.tsx:11]: fixed

File: app/components/ellipsisDropdownMenu.tsx [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
     const dropdownSelectedItemClass = (isActive: boolean) => {
         return isActive
             ? "px-4 py-2 block hover:text-tiffany hover:cursor-pointer text-sm"
             : "text-gray-900 block px-4 py-2 text-sm";
     };
     return (
-        <Menu as="div" className="relative inline-block text-left z-20">
-            <Menu.Button className="flex items-center text-white hover:text-gray-300 p-2 rounded-full ellipsismenu focus:outline-none" />
+        <Menu as="div" className="relative inline-block text-left z-15">
+            <MenuButton className={`flex items-center ${ellipsisENUM[location]} hover:text-gray-300 p-2 rounded-full ellipsismenu focus:outline-none`} />
```
[the remark this answers is not in the document]
Comment (HankC138) [reply app/components/ellipsisDropdownMenu.tsx:22]: done

File: app/components/navbar.tsx [hunk 4]
```diff
@@ -43,18 +43,31 @@ const Navbar: React.FC = () => {
 		},
 		{ href: "/projects", label: "Projects", isActive: pathname === "projects" },
 	];
-	const ellipsisLinks = [
+	const ellipsisDropdownMenuOption = (href: string, label: string) => {
+		return <Link href={href}>{label}</Link>;
+	};
+	const ellipsisDropdownMenuOptions = [
+		{
+			component:()=> ellipsisDropdownMenuOption(
```
[the remark this answers is not in the document]
Comment (HankC138) [reply app/components/navbar.tsx:51]: for some reason when I did it the first time that didn't work but now that I've tried again it works, the function has been removed thank you

**Comment by HankC138** [comment]
I believe I got that fixed according to your suggestions @zhukovdigital

**Pull request merged**